core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -474,13 +477,13 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * Spark fair scheduler pool.
    */
   def setLocalProperty(key: String, value: String) {
-    if (localProperties.get() == null) {
-      localProperties.set(new Properties())
+    if (localProperties.get().isEmpty) {
+      localProperties.set(Some(new Properties()))
```

Comment:
Basically, this behavior of having `localProperties` be null or None until we set at least one property is confusing to me; I think it's simpler to just eagerly perform this initialization before we set any properties.